backend/engine/plugins/node_dependencies/parse.py

```diff
@@ -111,15 +111,15 @@ def _find_versions_v2(lockfile: dict, nodes: list, resolver: LineNumberResolver)
     for node in nodes:
         pkg = lockfile["packages"][node]
         path = f"/{node}".replace("/node_modules/", ">")[1:]
-        source = resolver.find(pkg["integrity"]) or resolver.find(path)
+        source = resolver.find(pkg.get("integrity")) or resolver.find(path)
         versions.append(
             {
                 "path": path,
                 "version": pkg["version"],
-                "integrity": pkg["integrity"],
+                "integrity": pkg.get("integrity", ""),
```

Comment:
Is empty string the best value for empty here? How would the rest of Artemis handle empty string as opposed to something like "None provided"?

(Not suggesting we change it here, just trying to understand. This also applies to `filename` below)

Reply:
If you go up in this file, we never do anything with the integrity field, so it shouldn't change anything.